src/validation/fields/date.validation.ts

```diff
@@ -413,3 +452,86 @@ export const dateBecameIPLegalEntityBeneficialOwner = conditionalDateValidations
+  ...conditionalDateValidations(trustIndividualCeasedDateValidationsContext),
+
+  body("ceasedDate")
+    .if((value, { req }) => {
+      // check they are on an update or remove journey
+      // entity_number should have been populated by time they reach trust screens
+      const appData: ApplicationData = getApplicationData(req.session);
+      return !!appData.entity_number; // !! = truthy check
+    })
+    .if(body("stillInvolved").equals("0"))
+    .if(body("ceasedDateDay").notEmpty({ ignore_whitespace: true }))
+    .if(body("ceasedDateMonth").notEmpty({ ignore_whitespace: true }))
+    .if(body("ceasedDateYear").notEmpty({ ignore_whitespace: true }))
+    .custom((value, { req }) => {
+      checkCeasedDateOnOrAfterDateOfBirth(req, ErrorMessages.DATE_BEFORE_BIRTH_DATE_CEASED_TRUSTEE);
+
+      checkCeasedDateOnOrAfterTrustCreationDate(req, ErrorMessages.DATE_BEFORE_TRUST_CREATION_DATE_CEASED_TRUSTEE);
+
+      if (req.body["roleWithinTrust"] === RoleWithinTrustType.INTERESTED_PERSON
+        && req.body["dateBecameIPDay"]
+        && req.body["dateBecameIPMonth"]
+        && req.body["dateBecameIPYear"]) {
+        checkCeasedDateOnOrAfterInterestedPersonStartDate(req, ErrorMessages.DATE_BEFORE_INTERESTED_PERSON_START_DATE_CEASED_TRUSTEE);
+      }
+      return true;
+    })
+];
+
+const checkCeasedDateOnOrAfterInterestedPersonStartDate = (req, errorMessage: ErrorMessages) => {
+  checkFirstDateOnOrAfterSecondDate(
+    req.body["ceasedDateDay"], req.body["ceasedDateMonth"], req.body["ceasedDateYear"],
+    req.body["dateBecameIPDay"], req.body["dateBecameIPMonth"], req.body["dateBecameIPYear"],
+    errorMessage);
+};
+
+const checkCeasedDateOnOrAfterTrustCreationDate = (req, errorMessage: ErrorMessages) => {
+  const trustId = req.params ? req.params[ROUTE_PARAM_TRUST_ID] : undefined;
+  const appData: ApplicationData = getApplicationData(req.session);
+  const trust: Trust | undefined = getTrust(appData, trustId);
+  if (trust) {
+    checkFirstDateOnOrAfterSecondDate(
+      req.body["ceasedDateDay"], req.body["ceasedDateMonth"], req.body["ceasedDateYear"],
+      trust.creation_date_day, trust.creation_date_month, trust.creation_date_year,
+      errorMessage);
+  } else {
+    throw createAndLogErrorRequest(req, "Unable to find trust for ceased date validation");
+  }
+};
+
+const checkCeasedDateOnOrAfterDateOfBirth = (req, errorMessage: ErrorMessages) => {
+  checkFirstDateOnOrAfterSecondDate(
+    req.body["ceasedDateDay"], req.body["ceasedDateMonth"], req.body["ceasedDateYear"],
+    req.body["dateOfBirthDay"], req.body["dateOfBirthMonth"], req.body["dateOfBirthYear"],
+    errorMessage);
+};
+
+/**
+ * Try and find the trust based on trust id, if not provided then assume it is the trust marked as 'in review'
+ * @param appData
+ * @param trustId
+ * @returns Trust | undefined
+ */
+const getTrust = (appData: ApplicationData, trustId?: string): Trust | undefined => {
+  let trust: Trust;
```

Comment:
this block could be improved by using guard clauses

Reply:
We looked into the improvement but actually not quiet as straightforward as first thought. Plan to leave this as is.